### PR TITLE
[x] Improve code coverage generation (turn off sccache in code coverage by default, better warnings/logging, refactor)

### PR DESCRIPTION
**Pull request by rexhoffman**
## Motivation

Fix code coverage since the last rust update broke it.

### Have you read the [Contributing Guidelines on pull requests]

Yes

## Test Plan

Run by pushing code to gha-test-1 branch, results:
Runs:

https://github.com/diem/diem/runs/2814501836?check_suite_focus=true

Results:
https://ci-artifacts.diem.com/coverage/unit-coverage/2021-06-13-7ba68db8/index.html

## Related PRs

None

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

File: devtools/x/src/cargo.rs [hunk 1]
```diff
@@ -277,12 +277,14 @@ pub enum CargoCommand<'a> {
...
         args: &'a [OsString],
         env: &'a [(&'a str, Option<&'a str>)],
+        skip_sccache: &'a bool,
     },
     Build {
         cargo_config: &'a CargoConfig,
         direct_args: &'a [OsString],
         args: &'a [OsString],
         env: &'a [(&'a str, Option<&'a str>)],
+        skip_sccache: &'a bool,
```
Comment (bmwill) [inline devtools/x/src/cargo.rs:287]: `bool` is `Copy` so lets not take a reference to them but just have the type be `bool`

File: devtools/x/src/cargo.rs [hunk 2]
```diff
@@ -298,6 +300,14 @@ impl<'a> CargoCommand<'a> {
         }
     }
 
+    pub fn skip_sccache(&self) -> &bool {
```
Comment (bmwill) [inline devtools/x/src/cargo.rs:303]: Don't return a reference, just return `bool`

File: devtools/x/src/test.rs [hunk 3]
```diff
@@ -192,20 +220,23 @@ fn exec_lcov(html_lcov_path: &Path) -> Result<()> {
...
+        .arg("--binary-path")
         .arg(debug_dir.as_os_str())
         //source code location
         .arg("-s")
         .arg(project_root().as_os_str())
         //html output
         .arg("-t")
         .arg("html")
-        .arg("--llvm")
+        //        .arg("--llvm")
```
Comment (bmwill) [inline devtools/x/src/test.rs:241]: remove?

**Comment by rexhoffman** [comment]
prior build shows the line coverage test in test-code-coverage job failing when set to 90% for common/logger since the crate is only at 82%

**Review by bmwill — DISMISSED** [review]

**Pull request merged**